flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_Engine.js

```diff
@@ -69,11 +73,22 @@ var A320_Neo_LowerECAM_Engine;
             let n2Igniting = (SimVar.GetSimVarValue("TURB ENG IS IGNITING:1", "bool") === 1) ? true : false;
             let n2Percent = SimVar.GetSimVarValue("ENG N2 RPM:1", "percent");
             if (engineStarting && n2Igniting && n2Percent > 18 && n2Percent < 55) {
-                if (this.ignRightCurrentState == A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.A) {
-                    ignLeftTargetState = A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.B;
+                if (this.ignLeftOn != true) {
+                    if (this.ignLeftLastUsedIgniter == A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE) {
+                        ignLeftTargetState = A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.A;
+                    } else if (this.ignLeftLastUsedIgniter == A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.A) {
+                        ignLeftTargetState = A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.B;
+                    } else if (this.ignLeftLastUsedIgniter == A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.B) {
+                        ignLeftTargetState = A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.A;
+                    }
+                    this.ignLeftLastUsedIgniter = ignLeftTargetState;
+                    this.ignLeftOn = true;
```

Comment:
**thought**: A large part of the code is repeated for the left and right engine.
**suggestion**: Extract the code into a `function` which can then be used for either engine, or even better:
**suggestion**: Create a `class` (maybe something like `EngineIgnition`) which contains the newly added `isOn` and `lastUsedIgniter` and exposes a single function which returns the ignition state.

Reply:
Good suggestion, I created a new class that updates state. Was no need for isOn state as it can simply be inferred from the currentState and targetState != none.

Reply:
Great!